### ISSUE TITLE
Suport variable passing for include

### ISSUE THREAD

**Issue by epage** [cobalt-org/liquid-rust#310]
~~Depends on #275~~

Resources
- [Relevant tests](https://github.com/cobalt-org/liquid-rust/blob/66607e980263d9e704fb34a789e9146124e5629e/tests/conformance_ruby/tags/include_tag_test.rs) (did I tag the 237 when I meant 310?)
- [Rust code](https://github.com/cobalt-org/liquid-rust/blob/master/crates/lib/src/stdlib/tags/include_tag.rs)
- [Ruby code](https://github.com/Shopify/liquid/blob/master/lib/liquid/tags/include.rb)

**Comment by JosephLing** [comment]
Awesome library!

I have had a go at a hacked version of this as a custom include tag: https://gist.github.com/JosephLing/003e04607b09507d8e4d481316d6bb45

However it seems that the partial gets evaluated before `render_to` is called therefore I can't update the scope in time as such. Therefore would I be able to pick up this issue? and is there any pointers/help you might be able to give?

I think #275 is done now so it is okay to work on this?

**Comment by epage** [comment]
> I think #275 is done now so it is okay to work on this?

You are right.  There was a missing feature but I had already split it out into its own issue (#237), so I can close it out.

> However it seems that the partial gets evaluated before render_to is called therefore I can't update the scope in time as such. Therefore would I be able to pick up this issue? and is there any pointers/help you might be able to give?

The partial gets compiled before `render_to` but variable passing would be handled in `render_to`.  I'd recommend using `for` as your guide on this.  For example, it takes some [additional arguments](https://github.com/cobalt-org/liquid-rust/blob/master/crates/lib/src/stdlib/blocks/for_block.rs#L288) that are in a similar pattern to `include`.  It also demonstrates how to [set variables within a scope](https://github.com/cobalt-org/liquid-rust/blob/master/crates/lib/src/stdlib/blocks/for_block.rs#L175) like `include would.

I've also updated the links on the issue.
